contracts/test/executable/GMPExecutableWithTokenTest.sol

```diff
@@ -3,32 +3,40 @@
 pragma solidity ^0.8.0;
 
 import { AxelarGMPExecutableWithToken } from '../../executable/AxelarGMPExecutableWithToken.sol';
+import { IInterchainTransferReceived } from '../../interfaces/IAxelarGMPTransfer.sol';
 
-contract GMPExecutableWithTokenTest is AxelarGMPExecutableWithToken {
+contract GMPExecutableWithTokenTest is AxelarGMPExecutableWithToken, IInterchainTransferReceived {
     event Received(uint256 num);
     event ReceivedWithToken(uint256 num, address tokenAddress, uint256 amount);
 
     constructor(address gatewayAddress) AxelarGMPExecutableWithToken(gatewayAddress) {}
 
     function _execute(
-        bytes32, /*commandId*/
-        string calldata, /*sourceChain*/
-        string calldata, /*sourceAddress*/
+        bytes32 /*commandId*/,
+        string calldata /*sourceChain*/,
+        string calldata /*sourceAddress*/,
         bytes calldata payload
     ) internal override {
         uint256 num = abi.decode(payload, (uint256));
         emit Received(num);
     }
 
     function _executeWithToken(
-        bytes32, /*commandId*/
-        string calldata, /*sourceChain*/
-        string calldata, /*sourceAddress*/
+        bytes32 /*commandId*/,
+        string calldata sourceChain,
+        string calldata sourceAddress,
         bytes calldata payload,
         string calldata tokenSymbol,
         uint256 amount
     ) internal override {
-        uint256 num = abi.decode(payload, (uint256));
-        emit ReceivedWithToken(num, gatewayWithToken().tokenAddresses(tokenSymbol), amount);
+        // Emit InterchainTransferReceived event
+        emit InterchainTransferReceived(
```

Comment:
Technically, this is meant to be used for GMP execute, instead of GMP with token where the value is obvious. But we can help clean this up for you with an example in another PR